Project/Source/Components/ComponentTrail.cpp

```diff
@@ -99,9 +113,19 @@ void ComponentTrail::Save(JsonValue jComponent) const {
 		color++;
 	}
 	jComponent[JSON_TAG_NUMBER_COLORS] = trail->gradient->getMarks().size();
+
+	jComponent[JSON_TAG_TEXTURE_TEXTUREID] = trail->textureID;
+	JsonValue jFlip = jComponent[JSON_TAG_FLIP_TEXTURE];
+	jFlip[0] = trail->flipTexture[0];
+	jFlip[1] = trail->flipTexture[1];
+	jComponent[JSON_TAG_TEXTURE_REPEATS] = trail->nTextures;
+
+	jComponent[JSON_TAG_MAXVERTICES] = trail->maxVertices;
 }
+
 void ComponentTrail::Draw() {
-	trail->Draw();
+	float4x4 modelMatrix = float4x4::identity;
+	trail->Draw(modelMatrix);
```

Comment:
There are a few unnecessary copies in these two lines. You could pass the identity as a parameter directly to avoid this.